### ISSUE TITLE
Support Fabric components (new architecture)

### ISSUE THREAD

**Issue by benomatis** [stripe/stripe-react-native#1275]
I'm on RN 0.71.1 and have activated [the new architecture](https://reactnative.dev/docs/new-architecture-intro) and am receiving the message `'CardField' is not fabric compatible yet` in place of the actual `CardField` component.

Is it in the pipeline to make it compatible? This is only an issue in Android, in iOS it works fine.

**Comment by charliecruzan-stripe** [comment]
Yes it's in the pipeline, we'll be migrating all components + the module to turbomodules

**Comment by benomatis** [comment]
Cheers, Thank you @charliecruzan-stripe!

**Comment by smiley717** [comment]
![image](https://github.com/stripe/stripe-react-native/assets/80674675/bcd119b3-2630-48f4-b6d7-6d4cd950ac87)


I'm using RN 0.71.4 and when I'm trying to add stripe `CardFiled` There is an issue, Is this still pending?

**Comment by smiley717** [comment]
What is the solution, here? I cannot use react native stripe element? @charliecruzan-stripe

**Comment by benomatis** [comment]
@smiley717 sadly the only way to work around this is turn off the new architecture - not great, but it's the only way I could make `CardFiled` work. Again, this was only an issue in Android for me:

android/gradle.properties:

```env
# Use this property to enable support to the new architecture.
# This will allow you to use TurboModules and the Fabric render in
# your application. You should enable this flag either if you want
# to write custom TurboModules/Fabric components OR use libraries that
# are providing them.
newArchEnabled=false
# -> this here ^^^^^
```

**Comment by smiley717** [comment]
@benomatis  Thank you for your kind help, By turning it off this, it will affect to other rendering capabilities?

**Comment by smiley717** [comment]
Turning off this field is raising another compile issue. :/

**Comment by benomatis** [comment]
@smiley717 You'd have to test it for yourself; my app already existed before the new architecture came with 0.68, so I had no real impact turning it back off.

Your other compile issue may or may not be related to the new architecture, it can be many things, but it's not really the scope of this issue.

**Comment by smiley717** [comment]
@benomatis  thank you for collaborating. can you email me (on my github) for more details? As I think the another issue is not related to stripe, but lets go ahead now.

![image](https://github.com/stripe/stripe-react-native/assets/80674675/03fc81c2-2a8a-4c6d-8960-d82389033cfd)

Do you have any idea for this one?

**Comment by benomatis** [comment]
@smiley717 This wouldn't be sufficient information to troubleshoot something like this. I recommend you look further up for the exact error then search the internet for solutions, and post questions on one (or more) of the appropriate platforms, like https://stackoverflow.com/... Please let's not continue polluting this issue with unrelated comments. Thanks.

**Comment by smiley717** [comment]
Thanks a lot!

**Comment by smiley717** [comment]
![image](https://github.com/stripe/stripe-react-native/assets/80674675/05bbbd2a-8b10-46d3-b8f7-f59a702cf23b)

I was only able to fix this issue by disable `newArchEnabled` at the moment.

**Comment by sabuhiteymurov** [comment]
Is there any update on this? With the new architecture becoming the default standard, is it still not planned?

**Comment by amol-ghotankar** [comment]
New architecture is now default in react native so we should except company like stripe which is tech first payment company to solve these issues sooner and be the leadership in this space.

**Comment by ehtasham89** [comment]
@charliecruzan-stripe is there any update on when we can expect new version support?

**Comment by djMax** [comment]
Yes, this is really bad. Probably a hundred libraries in my project and *stripe* is the holdout breaking new architecture.

**Comment by sfriedman-stripe** [comment]
Hi all, thank you for sharing all of the feedback here. I agree with you; it's taken us way too long to support the new architecture. Fortunately, I have good news: we have begun our work to start migrating to the new architecture this week and hope to complete it over the next 1-2 months. If anything changes, you can expect to see an update from me here, but I'm optimistic we should be able to complete it by then.

I apologize again for the delay and look forward to getting this shipped for you all soon. Please keep the feedback coming as well; we will do a better job at communicating with you all in GitHub issues going forward.

**Comment by arsotech** [comment]
Hey guys, any ETA on the support of new architecture? We're all held back by this.

**Comment by sfriedman-stripe** [comment]
Hi @arsotech, nothing has changed with the timeline above. Stay tuned!

**Comment by arsotech** [comment]
@sfriedman-stripe thanks. My understanding from your comment above you were looking at 1 - 2 months back in early March. So should we expect it to be shipped in the next 2 - 3 weeks?

**Comment by sfriedman-stripe** [comment]
@arsotech Yes, we are currently targeting early May. I'll post here if anything changes

**Comment by flozero** [comment]
thats a bit sad to see the thread but hopefully you will meet your deadline

**Comment by irmakcosarsahna** [comment]
Stripe should resolve this issue very quickly.

**Comment by dreadrocksean** [comment]
As of today the AppStore, including Testflight only accepts an iOS 18.x minimum.  We’re now in no man's land as we need SDK 52 to build iOS 18 but … Stripe.  Any plans to pull the dealing closer? (Asking for a friend)

**Comment by goguda** [comment]
> As of today the AppStore, including TestFlight, only accepts an iOS 18.x minimum. We’re now in no man's land as we need SDK 52 to build iOS 18 but … Stripe. Any plans to pull the deadline closer? (Asking for a friend)

I think you’re misunderstanding the App Store requirement. They don’t only accept an iOS 18.x minimum, you just have to build against the iOS 18 SDK. Your minimum supported iOS version can still be less than iOS 18.

**Comment by erickreutz** [comment]
> As of today the AppStore, including TestFlight, only accepts an iOS 18.x minimum

I don't really understand what the problem is here. We have an app that's on the App Store that's on Expo 52 built with iOS 18 that uses the latest version of Stripe React Native. This is only about the new architecture, which is not a requirement to submit an app to the App Store.

**Comment by sfriedman-stripe** [comment]
> As of today the AppStore, including TestFlight, only accepts an iOS 18.x minimum. We’re now in no man's land as we need SDK 52 to build iOS 18 but … Stripe. Any plans to pull the deadline closer? (Asking for a friend)

The [deployment target](https://expo.dev/changelog/2024-11-12-sdk-52#highlights) for Expo 52 is iOS 15.1. Your deployment target is not the same as the minimum deployment setting.

**Comment by thunder6230** [comment]
Ppl just disable the new arch in the app.json

**Comment by dreadrocksean** [comment]
> > As of today the AppStore, including TestFlight, only accepts an iOS 18.x minimum. We’re now in no man's land as we need SDK 52 to build iOS 18 but … Stripe. Any plans to pull the deadline closer? (Asking for a friend)
> 
> The [deployment target](https://expo.dev/changelog/2024-11-12-sdk-52#highlights) for Expo 52 is iOS 15.1. Your deployment target is not the same as the minimum deployment setting. Apple is requiring the deployment target for your app to be iOS 18, not your minimum deployment.

Can I fulfill Apple’s requirement on Expo 51?

**Comment by sfriedman-stripe** [comment]
> Can I fulfill Apple’s requirement on Expo 51?

Let's keep the discussion in this issue focused on the Stripe SDK and the RN architecture. You can refer to the Apple developer forums to learn more about deployment targets.

**Comment by sfriedman-stripe** [comment]
Hi all, we're excited to share that support for the new architecture has been released today as v0.45.0 (fixed in #1893)! 🎉 Thank you again for your patience.

I will be closing this issue as resolved. If you experience any bugs with this release, **please open a new issue** as that will help with our tracking. We will not monitor bug reports that come in via _this_ issue after it is closed. We will try to get to new bug reports around the new architecture as soon as possible, as we know new architecture support is important to all of you.

**Comment by MartinCura** [comment]
fyi the README still has a warning saying you don't support it 😄 
https://github.com/stripe/stripe-react-native?tab=readme-ov-file#stripe-react-native-sdk
good work 👏🏼

**Comment by sfriedman-stripe** [comment]
@MartinCura Good flag, thank you! Will fix

**Comment by mattyx97** [comment]
@sfriedman-stripe why expo doctor say that?
```Running "expo doctor"
Running 15 checks on your project...
14/15 checks passed. 1 checks failed. Possible issues detected:
Use the --verbose flag to see more details about passed checks.
✖ Validate packages against React Native Directory package metadata
The following issues were found when validating your dependencies against React Native Directory:
  Unsupported on New Architecture: @stripe/stripe-react-native ````

i have 0.45.0 version installed

**Comment by arsotech** [comment]
> [@arsotech](https://github.com/arsotech) Yes, we are currently targeting early May. I'll post here if anything changes

@sfriedman-stripe any update on the new arch? Hoping for good news :)

**Comment by arsotech** [comment]
> Hi all, we're excited to share that support for the new architecture has been released today as v0.45.0 (fixed in [#1893](https://github.com/stripe/stripe-react-native/pull/1893))! 🎉 Thank you again for your patience.
> 
> I will be closing this issue as resolved. If you experience any bugs with this release, **please open a new issue** as that will help with our tracking. We will not monitor bug reports that come in via _this_ issue after it is closed. We will try to get to new bug reports around the new architecture as soon as possible, as we know new architecture support is important to all of you.

Well done thank you @sfriedman-stripe and the rn stripe team
